### PR TITLE
🐛 partial fix for amp-geo race using singleton service promise

### PR DESCRIPTION
**Pull request by jpettitt**
This is a partial fix for the `amp-geo` service race.  It move the service creation to script load time which greatly reduces the chances of it failing. A plan for a full fix in documented in #14841

### CODE REVIEW

File: extensions/amp-geo/0.1/amp-geo.js [hunk 1]
```diff
@@ -249,4 +248,20 @@ export class AmpGeo extends AMP.BaseElement {
   }
 }
 
-AMP.registerElement(TAG, AmpGeo);
+
+/**
+ * Create the service promise at load time to prevent race between extensions
+ */
+
+/** singleton {?Promise<!Object<string, (string|Array<string>)>>} */
```
Comment (jridgewell) [inline extensions/amp-geo/0.1/amp-geo.js]: `@type`

File: extensions/amp-geo/0.1/amp-geo.js [hunk 2]
```diff
@@ -249,4 +248,20 @@ export class AmpGeo extends AMP.BaseElement {
...
+let geoResolver = null;
+
+AMP.extension('amp-geo', '0.1', AMP => {
+  /** {Promise<!Object<string, (string|Array<string>)>>} */
```
Comment (jridgewell) [inline extensions/amp-geo/0.1/amp-geo.js]: `@type`

File: extensions/amp-geo/0.1/amp-geo.js (hunk 1 above)
Comment (jpettitt) [reply extensions/amp-geo/0.1/amp-geo.js]: doh!

File: extensions/amp-geo/0.1/amp-geo.js (hunk 2 above)
Comment (jpettitt) [reply extensions/amp-geo/0.1/amp-geo.js]: doh! * 2

**Pull request merged**